app/code/community/Paylike/Paymentmethod/Model/Paymentmethod.php

```diff
@@ -22,25 +23,44 @@ public function authorize(Varien_Object $payment, $amount)
     {
         $payment->setTransactionId($payment->getPaylikeTransactionId());
         $payment->setIsTransactionClosed(0);
-        return $this;
-
+        $paylike = new Paylike($this->getApiKey());
+        $result = $paylike->transactions->fetch($payment->getTransactionId());
+        if ($transaction = $result->transaction) {
+            if ($transaction->id != $payment->getTransactionId() || $transaction->amount != $amount * 100 || Mage::app()->getStore()->getCurrentCurrencyCode() != $transaction->currency) {
```

Comment:
`$transaction->id != $payment->getTransactionId()`

How does this work? Isn't this guaranteed to always be `true` because of line 24: `$payment->setTransactionId($payment->getPaylikeTransactionId());`?

You need to compare a unique reference from your records with something on the transactions `custom` field.